### DISCUSSION TITLE
Import CSS from within node_modules

### DISCUSSION THREAD

**Discussion by slavafomin** [vercel/next.js#24653] [Ideas]
### Describe the feature you'd like to request

Hello!

We have an internal UI components library in our company with around 100 components in it. Each component is distributed as a separate JavaScript file and a pre-built CSS file. In order for component to be easily consumed, the JS file has a CSS-import in it's header: `import './index.css'; export default Component …`. This works great with all our React applications.

However, some of our teams are using Next.js and it's behaving very opinionated on regard of CSS imports from within the `node_modules` directory, saying that it's not possible to do. I'm aware of the order import problem, but it doesn't affect us, because all of our component styles are generated using CSS modules with reasonable classnames to make overlapping impossible (we don't care about the import order and are perfectly willing to accept the undetermined/unstable import order).

I've tried to play with Webpack configuration today in order to overcome this problem, but I'm not happy with the results. First of all, Next.js is falling back to using webpack 4 instead of webpack 5 when the custom configuration is present. I'm also getting the notification: "Built-in CSS support is being disabled" and we don't want that as well.

So we have found ourselves in situation when Next.js can't be used in our ecosystem due to it's opinionated behavior and at the same time very fragile build configuration system. Refactoring all our 100 components and numerous existing applications  would also be unjustified just to support small number of Next.js applications. Also, this will degrade the end-developer experience because she will need to manually import CSS files for all the components used on each page of the application. And we need to consider the fact that one component could be built on top of 10 other components, so in order to import it you have to import 11 CSS files as well, which is just crazy.

### Describe the solution you'd like

The Next.js should provide an option to opt-out of this opinionated behavior for developers who understand and are willing to take the risks and/or it should provide a more robust (non-fragile) build customization system.

### Describe alternatives you've considered

At this point we can't use Next.js in our ecosystem and changing the ecosystem to support Next.js in just unreasonable. I can't see any other options, but I'm very open for suggestions. Thank you.

**Comment by joaogarin** [comment]
I have the same problem, but actually I only have Next.js applications. I kind of wonder, how are people publishing shared component libraries to npm using css modules to be consumed in nextjs? Building a giant css file with all styles of all components doesn't seem reasonable (like at all).

Any ideas or repositories showing how to publish components that can be consumed in nextjs that are not like "pre-bundling" all the css in the library itself? I kind of want that to be done in the Next.js application as that is where the components are consumed..and want to generate only CSS for the components that are used 🤔 (even if it generates all css in one file, if only for that application that would be fine)

I might be missing something..but this seems really not obvious to me at least.
